extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/interceptor/TransactionalInterceptorBase.java

```diff
@@ -104,18 +111,21 @@ protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm) thro
     protected Object invokeInOurTx(InvocationContext ic, TransactionManager tm, RunnableWithException afterEndTransaction)
             throws Exception {
 
-        TransactionConfiguration configAnnotation = getTransactionConfiguration(ic);
+        Integer timeoutConfiguredForMethod = cache.computeIfAbsent(ic.getMethod(),
+                (m) -> extractTransactionConfigurationTimeoutFromAnnotation(ic));
+
         int currentTmTimeout = ((CDIDelegatingTransactionManager) transactionManager).getTransactionTimeout();
-        if (configAnnotation != null && configAnnotation.timeout() != TransactionConfiguration.UNSET_TIMEOUT) {
-            tm.setTransactionTimeout(configAnnotation.timeout());
+
+        if (Objects.nonNull(timeoutConfiguredForMethod)) {
+            tm.setTransactionTimeout(timeoutConfiguredForMethod);
         }
+
         Transaction tx;
         try {
             tm.begin();
             tx = tm.getTransaction();
         } finally {
-            if (configAnnotation != null && configAnnotation.timeout() != TransactionConfiguration.UNSET_TIMEOUT) {
-                //restore the default behaviour
+            if (Objects.nonNull(timeoutConfiguredForMethod)) {
```

Comment:
Maybe I'm old school but I do prefere `timeoutConfigureForMethod != null` then using Objects here that didn't have any readability advantages.